presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java

```diff
@@ -978,6 +979,35 @@ public Void visitRevoke(Revoke node, Integer indent)
             return null;
         }
 
+        @Override
+        public Void visitShowGrant(ShowGrant node, Integer indent)
+        {
+            builder.append("SHOW GRANT ");
+            if (node.getIdentityType().isPresent()) {
+                if (node.getIdentityType().get() == ShowGrant.IdentityType.USER) {
+                    builder.append("USER ")
+                            .append(node.getIdentity().get())
+                            .append(" ");
+                }
+                else {
+                    throw new UnsupportedOperationException("unhandled identity : " + node.getIdentityType());
```

Comment:
I think this exception is to catch implementation errors, and not end user syntax errors. So, the information that this exception is coming from `visitShowGrant()` method should be sufficient. (I followed other methods in this class that throw exceptions such as `visitJoin` etc.)